outline.md

```diff
@@ -1,10 +1,55 @@
+* [*Overview*](https://github.com/spack/spack/discussions/20256#discussioncomment-248938)
+* *Definitions all conform to the ones in [Workflows](https://spack.readthedocs.io/en/latest/workflows.html#package-concrete-spec-installed-package).*
+
+# Use Cases
+* Concepts in this doc should map back to a specific requirement from some use case:
+  1. spack *user*
+  1. spack *package developer*
+  1. spack *maintainer*
+* A single person may represent any of these use cases at different moments.
+* [Workflows](https://spack.readthedocs.io/en/latest/workflows.html) will implicitly involve input from all of these use cases.
+
+The below example attempts to demonstrate how users, package developers, and maintainers interact in a **non-Spack context**:
+
+## Example: Pip Floating Constraints
```

Comment:
This section has a lot of good stuff in it but it's written in a way that's *very* specific to a particular package manager -- we can use pip as an example for sure, but not as a section header.

Reply:
Yes, agreed.